resource-service/src/contract-tests/java/uk/gov/gchq/palisade/contract/resource/kafka/KafkaInitalizer.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2021 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.palisade.contract.resource.kafka;
+
+import akka.actor.ActorSystem;
+import akka.stream.Materializer;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.admin.NewTopic;
+import org.apache.kafka.common.serialization.Deserializer;
+import org.apache.kafka.common.serialization.Serializer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.context.ApplicationContextInitializer;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Primary;
+import org.springframework.core.env.Environment;
+import org.springframework.core.io.ResourceLoader;
+import org.springframework.core.serializer.support.SerializationFailedException;
+import org.springframework.test.context.support.TestPropertySourceUtils;
+import org.testcontainers.containers.KafkaContainer;
+
+import uk.gov.gchq.palisade.service.resource.model.AuditErrorMessage;
+import uk.gov.gchq.palisade.service.resource.stream.PropertiesConfigurer;
+
+import java.io.IOException;
+import java.util.AbstractMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Stream;
+
+import static java.util.stream.Collectors.toMap;
+import static org.apache.kafka.clients.admin.AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG;
+
+class KafkaInitializer implements ApplicationContextInitializer<ConfigurableApplicationContext> {
+    private static final Logger LOGGER = LoggerFactory.getLogger(KafkaInitializer.class);
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+
+
+    static KafkaContainer kafka = new KafkaContainer("5.5.1")
```

Comment:
Kafka container can be final
static final KafkaContainer KAFKA_CONTAINER = new KafkaContainer("5.5.1")
            .withReuse(true);

Reply:
Updated